File: python/categorized_items/normal_item.py
# -*- coding: utf-8 -*-
from categorized_item import CategorizedItem

class NormalItem(CategorizedItem):
    def __init__(self, item):
        super().__init__(item)

    def update_sell_in(self):
        self.decrease_sell_in()

    def update_quality(self):
        if self.sell_in < 0:
            self.decrease_quality()
        self.decrease_quality()

File: cpp/src/categorized_items/NormalItem.cc
#include "NormalItem.h"

NormalItem::NormalItem(Item &item) : CategorizedItem(item){}

void NormalItem::updateQuality(){
    if (CategorizedItem::getSellIn() < 0) {
        CategorizedItem::decrementQuality();
    }
    CategorizedItem::decrementQuality();
}
void NormalItem::updateSellIn(){
    CategorizedItem::decrementSellIn();
}